### DISCUSSION TITLE
Hi, what's the difference in these two ways.

### DISCUSSION THREAD

**Discussion by c0sc0s** [styled-components/styled-components#4295] [Q&A] [answered]
1:
```js
style.div`
  .list a {
     &::after {
      
     }
  }
`
```
2:
```js
style.div`
  .list a {
     ::after {

     }
  }
`
```

At first time, i used the second way in my project, and it's unuseful, but when i change it to the first way, it's run.
But i can't understand why?
plz.

**Comment by Foxeian** [comment]
Pseudo-selectors in `styled-components` works just like in CSS. (or rather, Sass)

Here is an example of how to use a simple :after:
```
const HelloDiv = styled.div`
  &:after {
    content: "👋";
  }
`;

<HelloDiv>Hello</HelloDiv> // renders: "Hello 👋"
```
If the styled component is the same as the target element. You can do the following:
```
const Hello = styled.a`
  &:hover:before {
    width: 100%;
  }
`;
```
If you are targeting an element nested inside the styled component:
```
const Hello = styled.div`
  a:hover:before {
    width: 100%;
  }
`;
```

**Comment by Choi-Jiwon-38** [comment]
The reason the first method works and the second one doesn't is because in the first method, &::after refers to the full selector, including the parent selector. The `&` symbol represents the current selector (in this case, .list a), and ::after is added to it.

```css
.list a::after {
  /* styles */
}
```

 In the second method, there's no `&` before ::after, meaning it doesn’t connect with the parent selector. As a result, the browser doesn’t recognize the second approach, and the styles aren’t applied.

 In short, &::after ensures that the selector is correctly compiled as .list a::after, which is why the first approach works.

**Comment by quantizor** [comment, marked as answer]
The `&` character in styled-components (and Sass/SCSS) refers to the parent selector. Inside a nested block, `&::after` means "the parent element's `::after` pseudo-element", while `::after` without `&` is treated as a descendant selector — it looks for a `::after` inside the parent, which is not valid CSS.

So `&::after` compiles to `.generatedClass::after` (correct), while bare `::after` compiles to `.generatedClass ::after` (incorrect — note the space).

Always use `&` when targeting pseudo-elements and pseudo-classes within nested rules.
